### PR TITLE
Added AzureAutomation logger

### LINKED ISSUE #15: Log Harvester
Log harvester should be created that harvests logs into a storage account

### PR DESCRIPTION
**Pull request by lfshr**
Added a module for Azure Automation logging. This closes #15.

### CODE REVIEW

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            foreach ($Entry in $StreamData) {
+                $RecordParams = @{
+                    Id = $Entry.StreamRecordId
+                    JobId = $Job.JobId
+                    ResourceGroupName = $ResourceGroupName
+                    AutomationAccountName = $AutomationAccountName
+                }
+                $Record = Get-AzureRmAutomationJobOutputRecord @RecordParams
+                
+            $StreamOutput += @{
```
Comment (wibblemonkey) [inline src/Logging/AzureAutomationTools.Logging.psm1]: Misaligned line.

**Review by wibblemonkey — CHANGES_REQUESTED** [review]
Looks fine to me except for that one misaligned line. Good stuff.

File: src/Logging/AzureAutomationTools.Logging.psm1 (hunk 1 above)
Comment (lfshr) [reply src/Logging/AzureAutomationTools.Logging.psm1]: Fixed in latest commit - Formatted document.

**Review by wibblemonkey — APPROVED** [review]

**Comment by lfshr** [comment]
@stvnrs there are a few changes to your stuff here so I'll wait for your review before merging

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    foreach ($Job in $Jobs) {
+        Write-Verbose "Capturing:@{ Name = $($Job.RunbookName); Start = $($Job.CreationTime.UtcDateTime.ToString('s')) }"
+        $JobName = $Job.RunbookName
+        $CreateDate = $job.CreationTime.UtcDateTime
+        $Year = $CreateDate.Year
+        $Month = [string]::Format('{0:00}', $CreateDate.Month)
+        $Day = [string]::Format('{0:00}', $CreateDate.Day)
+        $JobBlobPath = "$ResourceGroupName/$AutomationAccountName/$JobName/$Year/$Month/$Day/$($Job.JobId).json"
+
+        try {
```
Comment (stvnrs) [inline src/Logging/AzureAutomationTools.Logging.psm1:150]: This should be optional - quite dangerous as a default

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 3]
```diff
@@ -0,0 +1,224 @@
+<#
+.Synopsis
+   Formats Azure Automation Runbook logs as JSON and exports to a storage account
+.DESCRIPTION
+   Formats Azure Automation Runbook logs as JSON and exports to a storage account
+.EXAMPLE
+   TODO
+#>
+function Export-AzureRmAutomationRunbookLog {
```
Comment (stvnrs) [inline src/Logging/AzureAutomationTools.Logging.psm1]: Should support ShouldProcess in next version

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+        # Runbook names to exclude
+        [Parameter()]
+        [string[]]
+        $ExcludeRunbook = $null,
+
+        # Optional Azure EnvironmentName to run this on. Storage account and automation account both need to be in this environment! 
+        [Parameter()]
+        [string]
+        $EnvironmentName = 'AzureCloud',
```
Comment (stvnrs) [inline src/Logging/AzureAutomationTools.Logging.psm1:42]: Remove this param - should not login.

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        throw "CollectLogsFrom cannot be greater than CollectLogsTo! CollectLogsFrom: $($CollectLogsFrom.ToString('s')); CollectLogsTo: $($CollectLogsTo.ToString('s'))"
+    }
+
+    if ($null -eq $IncludeRunbook) {
+        $IncludeRunbook = '*'
+    }
+
+    #endregion
+
+    try {
```
Comment (stvnrs) [inline src/Logging/AzureAutomationTools.Logging.psm1]: Remove

File: src/Logging/AzureAutomationTools.Logging.psm1 [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        $LogObjParams | ConvertTo-Json -Depth 100 | Set-Content -Path $LogPath -Force
+
+        Set-AzureStorageBlobContent -File $LogPath -Container $StorageContainerName -Blob $JobBlobPath -Context $StorageContext -ErrorAction Stop | Out-Null
+        Write-Output "Successfully uploaded log: '$JobBlobPath'"
+        Remove-Item -Path $LogPath -Force
+    }
+
+
+    Write-Output "Finished harvesting logs."
+}
```
Comment (stvnrs) [inline src/Logging/AzureAutomationTools.Logging.psm1]: Needs a new line

**Review by stvnrs — CHANGES_REQUESTED** [review]
Changes required - see comments

**Review by stvnrs — APPROVED** [review]

File: src/Logging/AzureAutomationTools.Logging.psm1 (hunk 2 above)
Comment (lfshr) [reply src/Logging/AzureAutomationTools.Logging.psm1:150]: Hard error in commit tree. Future release may make this optional.

File: src/Logging/AzureAutomationTools.Logging.psm1 (hunk 4 above)
Comment (lfshr) [reply src/Logging/AzureAutomationTools.Logging.psm1:42]: Done.

File: src/Logging/AzureAutomationTools.Logging.psm1 (hunk 6 above)
Comment (lfshr) [reply src/Logging/AzureAutomationTools.Logging.psm1]: Done.

**Pull request merged**